pkg/build/builder/dockerutil.go

```diff
@@ -39,15 +47,31 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 		opts.OutputStream = os.Stderr
 	}
 	var err error
-	for retries := 0; retries <= DefaultPushRetryCount; retries++ {
-		err = client.PushImage(opts, authConfig)
+
+	for retries := 0; retries <= MaxPushRetryCount; retries++ {
+
+		err := client.PushImage(opts, authConfig)
 		if err == nil {
 			return nil
 		}
-		if retries == DefaultPushRetryCount {
+
+		errMessage := fmt.Sprintf("%s", err)
+		var errInfo string
+
+		switch true {
+		case strings.Contains(errMessage, "v1 ping attempt failed with error"):
+			if retries > DefaultPushRetryCount {
+				return err
+			}
+			errInfo = " cause Docker registry is unavailable,"
+		case strings.Contains(errMessage, "is already in progress"):
+			errInfo = " cause other image is already being pushed to registry,"
+		default:
 			return err
 		}
-		util.HandleError(fmt.Errorf("push for image %s failed, will retry in %s ...", name, DefaultPushRetryDelay))
+
+		util.HandleError(fmt.Errorf("push for image %s failed,%s will retry in %s ...", name, errInfo, DefaultPushRetryDelay))
```

Comment:
> nm, as long as util.HandleError is always outputting the error string.

@bparees you mean to have both HandleError and glog ?

> For the default cases (the one that we are not checking in the switch) I dont set the errInfo, just those that we recognized.

i understand that but i don't know why you'd think those would start with whitespace?

anyway i agree with @mfojtik that we shouldn't really be hiding the original error.  Adding descriptive info around it is ok, but we should still expose the underlying error.